egui_glow/src/epi_backend.rs

```diff
@@ -145,7 +155,12 @@ pub fn run(app: Box<dyn epi::App>, native_options: &epi::NativeOptions) -> ! {
                 }
 
                 integration.on_event(&event);
-                if integration.should_quit() {
+
+                if EpiQuit::Recoverable == integration.should_quit() {
+                    if integration.on_pre_exit() {
+                        *control_flow = glutin::event_loop::ControlFlow::Exit;
+                    }
+                } else if EpiQuit::Unrecoverable == integration.should_quit() {
```

Comment:
Instead of duplicating all this code, I think it should be possible for `EpiIntegration` to handle it and have the `egui_glow` and `egui_glium` code the same as on `main`.

That is: `EpiIntegration::on_event` and `EpiIntegration::update` can just call `app.on_pre_exit` directly. I don't think there is any need for `EpiQuit`.